scripts/process.py

```diff
@@ -0,0 +1,161 @@
+"""
+    Retrieves stock market data and financials for the S&P 500
+    Reads ..data/constituents.csv and writes ../data/constiuents-financials.csv
+
+    Stock API are come and go.
+    Presently, this uses the IEX Trading API by IEX Group, Inc.
+    Data provided for free by IEX. https://iextrading.com/developer
+"""
+
+from dataflows import Flow, PackageWrapper, ResourceWrapper, validate
+from dataflows import add_metadata, dump_to_path, load, set_type, printer
+from collections import OrderedDict
+from os import mkdir
+from os.path import exists
+from bs4 import BeautifulSoup
+import urllib.request
+import json
+import csv
+
+
+EDGAR_BASE_URL = "http://www.sec.gov/cgi-bin/browse-edgar?action=getcompany&CIK="
+
+IEX_NUMBER_OF_SYMBOLS_PER_REQUEST = 100
+IEX_BASE_URL = "https://api.iextrading.com/1.0/stock/market/batch"
+IEX_TYPES_FIELDNAMES_TITLES = OrderedDict({
+    "quote": [
+        ["close", "Price"],
+        ["peRatio", "Price/Earnings"]
+    ],
+    "stats": [
+        ["dividendYield", "Dividend Yield"],
+        ["latestEPS", "Earnings/Share"],
+        ["week52high", "52 Week Low"],
+        ["week52low", "52 Week High"],
+        ["marketcap", "Market Cap"],
+        ["EBITDA", "EBITDA"],
+        ["priceToSales", "Price/Sales"],
+        ["priceToBook", "Price/Book"]
+    ]
+})
+
+
+def rename(package: PackageWrapper):
+    package.pkg.descriptor['resources'][0]['name'] = 's-p-constituents'
+    package.pkg.descriptor['resources'][0]['path'] = 'data/constituents.csv'
+    package.pkg.descriptor['resources'][1]['name'] = 'constiuents-financials'
+    package.pkg.descriptor['resources'][1]['path'] = 'data/constiuents-financials.csv'
+    yield package.pkg
+    res_iter = iter(package)
+    first: ResourceWrapper = next(res_iter)
+    second: ResourceWrapper = next(res_iter)
+    yield first.it
+    yield second.it
+    yield from package
+
+
+def extract_constituents(html):
+    datadir = 'data'
+    if not exists(datadir):
```

Comment:
Why are you creating this directory?

Reply:
I forgot that the dataflows does this automatically. I fixed it